### ISSUE TITLE
JSON Response - Mark Validation Failed

### ISSUE THREAD

**Issue by nicky-johnstone** [qcode-software/qcode-tcl#193]
We require some way to mark the validation process as failed outside of relying on the record items like we currently do.

`qc::record all_valid` is the check performed as to whether validation passes or fails while using `filter_validate`. It simply checks each record item and if any item is invalid  then validation has failed. There are occurrences where validation might fail for reasons that are not present in the record and we require a way of specifying this failure.

### Ideas
* Use an existing object (message or action) to indicate failed validation.
* Create a new object that indicates failed validation.

**Possible Names**

* invalid
* validation
* problem
* input
* user_error
* issue
* obstacle
* trouble

### Examples
**Message**
```json
{
    "record": {
    },
    "message": {
        "invalid": {
            "reason": "Reason validation failed."
        }
    },
    "action": {
    }
}
```

**Action**
```json
{
    "record": {
    },
    "message": {
    },
    "action": {
        "invalid": "Reason validation failed."
    }
}
```

**Separate Object**
```json
{
    "record": {
    },
    "message": {
    },
    "action": {
    },
    "invalid": {
        "reason": "Reason validation failed."
    }
}
```

**Comment by bvw** [comment]
What reasons would there be for "no-can-do"

Validation failed for some reason
Permission not allowed
Internal Server Error

any others ?

Wondering if we should have a "separate object" top level `status:valid or invalid`
What about the reasons - could they be one or info alert notify error ?

**Comment by nicky-johnstone** [comment]
I think that sums it up in terms of reasons for failure.

Some more examples:

**Stand-alone**
```json
{
    "status": {
        "valid": false,
        "reason": "alert",
        "message": "Bunch of reasons why validation failed..."
    }
}
```

```json
{
    "invalid": {
        "reason": "Bunch of reasons why it failed..."
    }
}
```

```json
{
    "invalid": {
        "reason": "error",
        "message": "Software bug..."
    }
}
```

**Combination**
Use new item to outline status and the `message` object to detail why.
```json
{
    "status": "invalid",
    "message": {
        "alert": {
            "value": "The user account is not open."
        }
    }
}
```
N.B. We can probably change the Global JSON Response to drop the `value` portion of the `message` object at some point  i.e.

```json
{
    "message": {
        "alert": "Alerting the user...",
        "error": "The boffins made a mistake.",
        "notify": "Password is due to be changed in 1 day."
    }
}
```

**Comment by bvw** [comment]
okay so global property of the top level object can be `status:'whatever'` and messages can be independent of that ?

Just want to emphasise this:-
https://felipec.wordpress.com/2013/10/07/the-linux-way/

We are at an early stage but as we implement we will be much more restricted with scope to change.

**Comment by nicky-johnstone** [comment]
Merged in on pull request #194
